pkg/utils/resource/update/update.go

```diff
@@ -0,0 +1,72 @@
+package update
+
+import (
+	"context"
+	"reflect"
+
+	cco "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/aws-load-balancer-operator/pkg/utils"
+)
+
+// CredentialsRequest updates the given CredentialsRequest to the desired state if needed.
+// Returns true if the CredentialsRequest was updated, false otherwise.
+func CredentialsRequest(ctx context.Context, client client.Client, current *cco.CredentialsRequest, desired *cco.CredentialsRequest) (bool, error) {
+	if changed, err := IsCredentialsRequestChanged(current, desired); err != nil {
+		return false, err
+	} else if !changed {
+		return false, nil
+	}
+
+	updated := current.DeepCopy()
+	updated.Name = desired.Name
+	updated.Namespace = desired.Namespace
+	updated.Spec = desired.Spec
+	if err := client.Update(ctx, updated); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// IsCredentialsRequestChanged compares the current and desired states of a CredentialsRequest.
+// Returns true of current start is different from the desired, false otherwise.
+func IsCredentialsRequestChanged(current, desired *cco.CredentialsRequest) (bool, error) {
+	if current.Name != desired.Name {
```

Comment:
Like I mentioned above, I don't think name and namespace is ever possible to be different. I am missing something?